bindings/c/test/mako/mako.c

```diff
@@ -801,28 +908,73 @@ void* worker_thread(void* thread_args) {
 
 	/* build/popualte */
 	else if (args->mode == MODE_BUILD) {
-		rc = populate(transaction, args, worker_id, thread_id, thread_tps, stats);
+		rc =
+		    populate(transaction, args, worker_id, thread_id, thread_tps, stats, block, elem_size, is_memory_allocated);
 		if (rc < 0) {
 			fprintf(stderr, "ERROR: populate failed\n");
 		}
 	}
 
 	/* run the workload */
 	else if (args->mode == MODE_RUN) {
-		rc = run_workload(transaction, args, thread_tps, throttle_factor, thread_iters, signal, stats, dotrace);
+		rc = run_workload(transaction, args, thread_tps, throttle_factor, thread_iters, signal, stats, dotrace, block,
+		                  elem_size, is_memory_allocated);
 		if (rc < 0) {
 			fprintf(stderr, "ERROR: run_workload failed\n");
 		}
 	}
 
+	if (args->mode == MODE_BUILD || args->mode == MODE_RUN) {
+		char str2[1000];
+		sprintf(str2, "/dev/shm/tempDataFile%d", *parent_id);
```

Comment:
Let's have this `/dev/shm/tempDataFile` string as a macro and define in a single place, maybe in `mako.h`.

Reply:
... hmm, or maybe we can parameterize this too.  `/dev/shm` is fast, but maybe you may want to use a real disk to reduce the memory pressure sometime.

Reply:
Yes, I will do this. It will make the code more neat.